pkg/controller/jenkins/notifications/smtp/smtp.go

```diff
@@ -0,0 +1,129 @@
+package smtp
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"strings"
+
+	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/event"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/provider"
+
+	"github.com/pkg/errors"
+	"gopkg.in/gomail.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	mailSubject = "Jenkins Operator Notification"
+
+	infoColor    = "blue"
+	warningColor = "red"
+	defaultColor = "gray"
+
+	content = `
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html>
+<head></head>
+<body>
+		<h1 style="background-color: %s; color: white; padding: 3px 10px;">%s</h1>
+		<h3>%s</h3>
+		<table>
+			<tr>
+				<td><b>CR name:</b></td>
+				<td>%s</td>
+			</tr>
+			<tr>
+				<td><b>Phase:</b></td>
+				<td>%s</td>
+			</tr>
+		</table>
+		<h6 style="font-size: 11px; color: grey; margin-top: 15px;">Powered by Jenkins Operator <3</h6>
+</body>
+</html>`
+)
+
+// SMTP is Simple Mail Transport Protocol used for sending emails
+type SMTP struct {
+	k8sClient k8sclient.Client
+	config    v1alpha2.Notification
+}
+
+// New returns instance of SMTP
+func New(k8sClient k8sclient.Client, config v1alpha2.Notification) *SMTP {
+	return &SMTP{k8sClient: k8sClient, config: config}
+}
+
+// Send is function for sending notification by SMTP server
+func (s SMTP) Send(e event.Event) error {
```

Comment:
I would separate the concerns of sending from message creation to simplify code for readability and testing